src/Plugin/TagMappingType/MediaEntity.php

```diff
@@ -32,22 +32,32 @@ public function optionsForm(array $form, FormStateInterface $form_state) {
 
     $options = [];
 
+    /** @var \Drupal\media_entity\MediaTypeInterface $bundle */
     foreach ($bundles as $bundle) {
-      $options[$bundle->id()] = $bundle->label();
+      if ($bundle->getType()->getPluginId() == 'image') {
+        $options[$bundle->id()] = $bundle->label();
+      }
     }
 
     $wrapper_id = Html::getId('tap-mapping-form-ajax-wrapper');
 
-    $form['bundle'] = [
-      '#type' => 'select',
-      '#title' => $this->t('Bundle'),
-      '#options' => $options,
-      '#default_value' => $this->getOption('bundle'),
-      '#ajax' => [
-        'callback' => '::ajaxCallback',
-        'wrapper' => $wrapper_id,
-      ],
-    ];
+    if (count($options) > 1) {
+      $form['bundle'] = [
+        '#type' => 'select',
+        '#title' => $this->t('Bundle'),
+        '#options' => $options,
+        '#default_value' => $this->getOption('bundle'),
+        '#ajax' => [
+          'callback' => '::ajaxCallback',
+          'wrapper' => $wrapper_id,
+        ],
+      ];
+    }
+    else {
+      $key = array_keys($options);
+      $this->setOptions($this->getOptions() + ['bundle' => reset($key)]);
```

Comment:
I personally would not hide the select in case there is only one option, but make the select a required element.
Additionally, when used, the type of the element should be `value` instead of `hidden`.